R/compat-purrr.R

```diff
@@ -50,7 +50,7 @@ pluck_cpl <- function(.x, .f) {
 }
 
 map2 <- function(.x, .y, .f, ...) {
-  mapply(.f, .x, .y, MoreArgs = list(...), SIMPLIFY = FALSE)
+  set_names(mapply(.f, .x, .y, MoreArgs = list(...), SIMPLIFY = FALSE), names(.x))
```

Comment:
I have a vague recollection that setting names unconditionally like this is a bit risky, but I can't remember why

Reply:
Perhaps because `set_names()` does not recycle whereas `mapply()` does? I think `set_names()` should recycle, it makes sense in some situations, and is handy for `set_names(x, "")`: https://github.com/r-lib/rlang/issues/325

However I see that that `purrr::map2()` discards the names when it recycles. So for consistency I'll check for length.